src/Security/Authorization/Policy/src/PolicyEvaluator.cs

```diff
@@ -38,19 +38,27 @@ public virtual async Task<AuthenticateResult> AuthenticateAsync(AuthorizationPol
             if (policy.AuthenticationSchemes != null && policy.AuthenticationSchemes.Count > 0)
             {
                 ClaimsPrincipal? newPrincipal = null;
+                DateTimeOffset? minExpiresUtc = null;
                 foreach (var scheme in policy.AuthenticationSchemes)
                 {
                     var result = await context.AuthenticateAsync(scheme);
                     if (result != null && result.Succeeded)
                     {
                         newPrincipal = SecurityHelper.MergeUserPrincipal(newPrincipal, result.Principal);
+
+                        if (minExpiresUtc is null || result.Properties?.ExpiresUtc < minExpiresUtc)
+                        {
+                            minExpiresUtc = result.Properties?.ExpiresUtc;
+                        }
                     }
                 }
 
                 if (newPrincipal != null)
                 {
                     context.User = newPrincipal;
```

Comment:
Should we still set the user here if we're just going to replace it with the new feature right after this call?

Reply:
I see no issues keeping it, also someone might call this outside of the normal flow